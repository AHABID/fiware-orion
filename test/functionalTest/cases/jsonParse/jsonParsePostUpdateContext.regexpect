1: ***********************************
HTTP/1.1 200 OK
Content-Length: 895
Content-Type: application/json
Date: REGEX(.*)

{
    "contextResponses": [
        {
            "contextElement": {
                "attributes": [
                    {
                        "metadatas": [
                            {
                                "name": "attributeValueIdentifier",
                                "type": "long",
                                "value": "1"
                            },
                            {
                                "name": "attributeValueIdentifier",
                                "type": "long",
                                "value": "2"
                            },
                            {
                                "name": "attributeValueIdentifier",
                                "type": "long",
                                "value": "3"
                            }
                        ],
                        "name": "temperature", 
                        "type": "degree",
                        "value": ""
                    }
                ], 
                "id": "ConferenceRoom", 
                "isPattern": "false", 
                "type": "Room"
            }, 
            "statusCode": {
                "code": "200", 
                "reasonPhrase": "OK"
            }
        }
    ]
}
2: ***********************************
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: REGEX(.*)

{
    "duration": "PT1M",
    "registrationId": "REGEX([0-9a-f]{24})"
}
3: ***********************************
HTTP/1.1 200 OK
Content-Length: 895
Content-Type: application/json
Date: REGEX(.*)

{
    "contextResponses": [
        {
            "contextElement": {
                "attributes": [
                    {
                        "metadatas": [
                            {
                                "name": "attributeValueIdentifier",
                                "type": "long",
                                "value": "1"
                            },
                            {
                                "name": "attributeValueIdentifier",
                                "type": "long",
                                "value": "2"
                            },
                            {
                                "name": "attributeValueIdentifier",
                                "type": "long",
                                "value": "3"
                            }
                        ],
                        "name": "temperature", 
                        "type": "degree",
                        "value": ""
                    }
                ], 
                "id": "ConferenceRoom", 
                "isPattern": "false", 
                "type": "Room"
            }, 
            "statusCode": {
                "code": "200", 
                "reasonPhrase": "OK"
            }
        }
    ]
}
4: ***********************************
