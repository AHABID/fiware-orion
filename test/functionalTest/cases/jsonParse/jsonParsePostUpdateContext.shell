echo "1: ***********************************"

url="/NGSI10/updateContext"
payload='{
  "contextElements": [
    {
      "type": "Room",
      "isPattern": "false",
      "id": "ConferenceRoom",
      "attributes": [
        {
          "name": "temperature",
          "type": "degree",
          "value": "20",
          "metadatas": [
            {
              "name": "attributeValueIdentifier",
              "type": "long",
              "value": "1"
            },
            {
              "name": "attributeValueIdentifier",
              "type": "long",
              "value": "2"
            },
            {
              "name": "attributeValueIdentifier",
              "type": "long",
              "value": "3"
            }
          ]
        }
      ]
    }
  ],
  "updateAction": "APPEND"
}'
curlJson ${url} "${payload}"

echo "2: ***********************************"

url="/NGSI9/registerContext"
payload='{
  "contextRegistrations": [
    {
      "entities": [
        {
          "type": "Room",
          "isPattern": "false",
          "id": "ConferenceRoom"
        },
        {
          "type": "Room",
          "isPattern": "false",
          "id": "OfficeRoom"
        }
      ],
      "attributes": [
        {
          "name": "temperature",
          "type": "degree",
          "isDomain": "false",
          "metadatas": [
            {
              "name": "ID",
              "type": "string",
              "value": "1110"
            }
          ]
        }
      ],
      "metadatas": [
        {
          "name": "ID",
          "type": "string",
          "value": "2212"
        }
      ],
      "providingApplication": "http://192.168.100.1:70/application"
    }
  ],
  "duration": "PT1M"
}'
curlJson ${url} "${payload}"

echo "3: ***********************************"
url="/NGSI10/updateContext"
payload='{
  "contextElements": [
    {
      "type": "Room",
      "isPattern": "false",
      "id": "ConferenceRoom",
      "attributes": [
        {
          "name": "temperature",
          "type": "degree",
          "value": "20",
          "metadatas": [
            {
              "name": "attributeValueIdentifier",
              "type": "long",
              "value": "1"
            },
            {
              "name": "attributeValueIdentifier",
              "type": "long",
              "value": "2"
            },
            {
              "name": "attributeValueIdentifier",
              "type": "long",
              "value": "3"
            }
          ]
        }
      ]
    }
  ],
  "updateAction": "UPDATE"
}'
curlJson ${url} "${payload}"

echo "4: ***********************************"
