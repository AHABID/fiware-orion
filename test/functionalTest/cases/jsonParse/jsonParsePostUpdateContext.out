1: ***********************************
HTTP/1.1 200 OK
Content-Length: 895
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:21:13 GMT

{
    "contextResponses": [
        {
            "contextElement": {
                "attributes": [
                    {
                        "metadatas": [
                            {
                                "name": "attributeValueIdentifier", 
                                "type": "long", 
                                "value": "1"
                            }, 
                            {
                                "name": "attributeValueIdentifier", 
                                "type": "long", 
                                "value": "2"
                            }, 
                            {
                                "name": "attributeValueIdentifier", 
                                "type": "long", 
                                "value": "3"
                            }
                        ], 
                        "name": "temperature", 
                        "type": "degree", 
                        "value": ""
                    }
                ], 
                "id": "ConferenceRoom", 
                "isPattern": "false", 
                "type": "Room"
            }, 
            "statusCode": {
                "code": "200", 
                "reasonPhrase": "OK"
            }
        }
    ]
}
2: ***********************************
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:21:13 GMT

{
    "duration": "PT1M", 
    "registrationId": "538c33f92d4b7ffb3396621f"
}
3: ***********************************
HTTP/1.1 200 OK
Content-Length: 895
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:21:13 GMT

{
    "contextResponses": [
        {
            "contextElement": {
                "attributes": [
                    {
                        "metadatas": [
                            {
                                "name": "attributeValueIdentifier", 
                                "type": "long", 
                                "value": "1"
                            }, 
                            {
                                "name": "attributeValueIdentifier", 
                                "type": "long", 
                                "value": "2"
                            }, 
                            {
                                "name": "attributeValueIdentifier", 
                                "type": "long", 
                                "value": "3"
                            }
                        ], 
                        "name": "temperature", 
                        "type": "degree", 
                        "value": ""
                    }
                ], 
                "id": "ConferenceRoom", 
                "isPattern": "false", 
                "type": "Room"
            }, 
            "statusCode": {
                "code": "200", 
                "reasonPhrase": "OK"
            }
        }
    ]
}
4: ***********************************
